src/api.jl

```diff
@@ -0,0 +1,188 @@
+- `allow_filename_change_dir::Bool`: Recommended to set to true as it could be a potential vulnerability. It forbids "/" in file names to prevent directory changes in Unix systems (default: `false`).
+
+## Returns:
+- `Julja2ParsedTemplate`: The parsed template object.
+"""
+function Julja2Template(template::String;
+                var_borders::BlockBorders=BlockBorders("{{", "}}"),
+                operator_borders::BlockBorders=BlockBorders("{%", "%}"),
+                templates_dir::String="./",
+                ignore_missing_vars=true,
+                allow_filename_change_dir=false)::Julja2ParsedTemplate
+
+    template_dir_path = _format_template_dir_path(templates_dir)
+    template_options = TemplateOptions(
+        var_borders, operator_borders, template_dir_path,
+        ignore_missing_vars, allow_filename_change_dir)
+    _validate_options(template_options)
+
+    root = Node("""function build_result_string()\n_result=IOBuffer()\n""", [])
+    build_tree!(template, 1, root, Vector{Julja2.TokenType}(), template_options)
+    generator_code = build_code(root)
+    generator_code *= "\nreturn String(take!(_result))\nend"
+    generator = Meta.parse(generator_code)
+
+    return Julja2ParsedTemplate(generator)        
+end
+
+function Julja2Template(template::Vector{UInt8};
+                    var_borders::BlockBorders=BlockBorders("{{", "}}"),
+                    operator_borders::BlockBorders=BlockBorders("{%", "%}"),
+                    templates_dir::String="./",
+                    ignore_missing_vars=true,
+                    allow_filename_change_dir=false)::Julja2ParsedTemplate
+
+    return Julja2Template(String(template),
+    var_borders=var_borders,
+    operator_borders=operator_borders,
+    templates_dir=templates_dir,
+    ignore_missing_vars=ignore_missing_vars,   
+    allow_filename_change_dir=allow_filename_change_dir)
+end
+
+macro julja2_template_str(content)
+    return :(Julja2Template($content))
+end
+
+"""
+    julja2_render(template::Julja2ParsedTemplate, obj) -> String
+
+Renders template from given precompiled template with obj values.
+
+## Arguments:
+- `template::String`: Parsed template.
+- `obj`: Is an object or Dict of arguments.
+
+## Returns:
+- `Julja2ParsedTemplate`: The parsed template object.
+
+## Usage:
+```julia
+using Julja2
+
+struct Book
+    title::String
+    authors::Vector{String}
+    year
+end
+
+template = Julja2.Julja2Template(\"\"\"
+###  {{ title}}
+{% if year != nothing %}Published in {{ year }}{% endif %}
+Authors:{% for author in authors %}
+- {{ author }}{% endfor %}\"\"\")
+print(julja2_render(template, Book("Coding 101", ["Mark", "Alice", "Bob"], 2024)))
+
+# Result: 
+\"\"\"
+### Coding 101
+Published in 2024
+Authors:
+- Mark
+- Alice
+- Bob
+\"\"\"
+```
+"""
+function julja2_render(template::Julja2ParsedTemplate, obj)::String
+    if typeof(obj) == Dict{String, Any}
```

Comment:
It is better to isolate such conditions into separate functions that rely on [multiple dispatch](https://docs.julialang.org/en/v1/manual/performance-tips/#Break-functions-into-multiple-definitions).